src/libfetchers/github.cc

```diff
@@ -29,7 +29,9 @@ std::regex hostRegex(hostRegexS, std::regex::ECMAScript);
 
 struct GitArchiveInputScheme : InputScheme
 {
-    virtual std::optional<std::pair<std::string, std::string>> accessHeaderFromToken(const std::string & token) const = 0;
+    virtual std::optional<std::pair<std::string, std::string>> accessHeaderFromToken(const std::string & token) const {
+        return std::pair<std::string, std::string>("Authorization", fmt("Bearer %s", token));
```

Comment:
Do we know which type of git hosters/software support this scheme? This works with sourcehut and who else?

Reply:
Sourcehut: https://lists.sr.ht/~sircmpwn/sr.ht-dev/%3C20200330195201.620-1-zash%40zash.se%3E
gitlab: https://gitlab.com/gitlab-org/gitlab/-/issues/340962
gitea: https://docs.gitea.com/next/development/api-usage#oauth2-provider

Reply:
Ok. This seems common enough!

Reply:
So what I am wondering is that you mentioned this would help with `git+https`. But I don't think this implementation is used for the git input scheme. So in which case is this patch helping?

Reply:
Ok, so I get a clearer picture: where is the handler for `git+https`?

Reply:
`fetchTree.cc` is what you are looking for.